mlx/coverity_services.py

```diff
@@ -1,614 +1,415 @@
+            list[str]: The list of valid attributes
+        """
+        logging.info("Validate required %s [%s]", name, req_csv)
         validated = ""
         delim = ""
-        for field in req_csv.split(','):
+        filter_values = []
+        for field in req_csv.split(","):
             if not valid_list or field in valid_list:
-                logging.info('Classification [%s] is valid', field)
-                filter_list.append(field)
+                logging.info("Classification [%s] is valid", field)
+                filter_values.append(field)
                 validated += delim + field
                 delim = ","
             elif allow_regex:
                 pattern = re.compile(field)
                 for element in valid_list:
-                    if pattern.search(element) and element not in filter_list:
-                        filter_list.append(element)
+                    if pattern.search(element) and element not in filter_values:
+                        filter_values.append(element)
                         validated += delim + element
                         delim = ","
             else:
-                logging.error('Invalid %s filter: %s', name, field)
-        return validated
+                logging.error("Invalid %s filter: %s", name, field)
+        return validated, filter_values
 
+    def assemble_query_filter(self, column_key, filter_values, matcher_type):
+        """Assemble a filter for a specific column
 
-class CoverityDefectService(Service):
-    '''
-    Coverity Defect Service (WebServices)
-    '''
-
-    def __init__(self, config_service):
-        '''Create a Defect Service, bound to the given Configuration Service'''
-        super(CoverityDefectService, self).__init__(config_service.get_transport(),
-                                                    config_service.get_hostname(),
-                                                    config_service.get_port(),
-                                                    config_service.get_ws_version())
-        self.config_service = config_service
-        self.filters = ""
-        # logging.getLogger('suds.client').setLevel(logging.DEBUG)
-        url = self.get_ws_url('defectservice')
-        self.validate_presence(url, 'Coverity Defect Service')
+        Args:
+            column_key (str): The column key
+            filter_values (list[str]): The list of valid values to filter on
+            matcher_type (str): The type of the matcher (nameMatcher, idMatcher or keyMatcher)
 
-    def get_defects(self, project, stream, filters, custom=None):
-        """ Gets a list of defects for given stream, with some query criteria.
+        Returns:
+            dict: New filter for API queries
+        """
+        matchers = []
+        for filter_ in filter_values:
+            matcher = {"type": matcher_type}
+            if matcher_type == "nameMatcher":
+                matcher["name"] = filter_
+                matcher["class"] = "Component"
```

Comment:
`"Component"` is the column name. Instead of passing the column key via `column_key`, I think it's safer to pass the column name instead and using `self.columns` to determine the column key based on it. 

See https://github.com/melexis/sphinx-coverity-extension/pull/74/files#r1714671877 and https://github.com/melexis/sphinx-coverity-extension/pull/74/files#r1714727518

Reply:
I will refactor the `self.columns` to a dict

Reply:
Resolved in aab78642dbefa9a8ec875efc7ee6dc8e9f555400 and edff3124515fecf965401f40c8555b6e829d3ef7